test/test_filters.rb

```diff
@@ -295,6 +295,13 @@ def select; end
         end
       end
 
+      context "with whitespace" do
+        should "return input" do
+          assert_nil(@filter.date_to_xmlschema(nil))
+          assert_equal("  ", @filter.date_to_xmlschema(""))
```

Comment:
It appears this assertion is invalid. Check out the CI output to see more. Looks like it returns `""`. Did you mean to pass `"  "` to `date_to_xmlschema`?

We should test all 3 methods you touched, too.